### PR TITLE
UCX: set dev_types

### PR DESCRIPTION
**Pull request by pleasantrabbit**
- set proper dev_types in Bind()
- update test_benchmark to use GPUs on the server

Signed-off-by: Yulu Jia <yulu.jia@bytedance.com>

### CODE REVIEW

File: src/ucx_van.h [hunk 1]
```diff
@@ -800,13 +802,31 @@ class UCXVan : public Van {
   int Bind(Node &node, int max_retry) override {
     contexts_.reserve(node.num_ports);
 
+    int num_cpu_dev   = GetEnv("DMLC_NUM_CPU_DEV", 1);
+    int num_gpu_dev   = GetEnv("DMLC_NUM_GPU_DEV", 0);
+
```
Comment (eric-haibin-lin) [inline src/ucx_van.h:807]: How do we typically set NUM_PORTS and DMLC_NUM_CPU_DEV/DMLC_NUM_GPU_DEV? Like
DMLC_NUM_GPU_DEV=NUM_PORTS=4 for 4 NICs?

Comment (pleasantrabbit) [reply src/ucx_van.h:807]: they should satisfy NUM_PORTS = DMLC_NUM_GPU_DEV + DMLC_NUM_CPU_DEV, and for the moment we assume there's only one type of devices

**Comment by pleasantrabbit** [comment]
@brminich

File: src/ucx_van.h (hunk 1 above)
Comment (brminich) [reply src/ucx_van.h:807]: will you need to support mixed types of device? if yes, why would need to maintain such restriction?

Comment (pleasantrabbit) [reply src/ucx_van.h:807]: yes we do want mixed types. this restriction is because the code doesn't support mixed dev types yet

Comment (eric-haibin-lin) [reply src/ucx_van.h:807]: Can we remove this restriction and test it again? @pleasantrabbit

Comment (pleasantrabbit) [reply src/ucx_van.h:807]: done. I removed the the check.

**Review by eric-haibin-lin — APPROVED** [review]

**Pull request merged**